src/sentry/eventstream/base.py

```diff
@@ -1,6 +1,8 @@
 import logging
-from typing import Literal, Optional, Union
+from datetime import datetime
+from typing import Any, Collection, Literal, Mapping, Optional, Sequence, Union
 
+from sentry.eventstore.models import BaseEvent
```

Comment:
Might need to import this `if TYPE_CHECKING`

Reply:
and to add `from __future__ import annotations` to the top of the file -- https://youtu.be/zH_F2xC0LOk

Reply:
I've wondered a few times if we should just bulk import that in every file

Reply:
I'm in favor of it -- there's a few quirks with things that (ab)use runtime typing like pydantic / etc. that we'd want to make sure we wouldn't anger

might throw together a branch to see how angy it is

Reply:
Thanks